From 6d9006bf1430e5f98cdddbacac9802d09fb0cd56 Mon Sep 17 00:00:00 2001
From: Paul Sauve <paul@burngames.net>
Date: Sun, 14 Jul 2019 20:51:08 -0500
Subject: [PATCH] If the Bukkit generator already has a spawn, use it
 immediately instead of spending time generating one that we won't use


diff --git a/src/main/java/net/minecraft/server/WorldServer.java b/src/main/java/net/minecraft/server/WorldServer.java
index ee071ba2..8af54019 100644
--- a/src/main/java/net/minecraft/server/WorldServer.java
+++ b/src/main/java/net/minecraft/server/WorldServer.java
@@ -769,12 +769,6 @@ public class WorldServer extends World implements IAsyncTaskHandler {
         } else if (this.worldData.getType() == WorldType.DEBUG_ALL_BLOCK_STATES) {
             this.worldData.setSpawn(BlockPosition.ZERO.up());
         } else {
-            WorldChunkManager worldchunkmanager = this.chunkProvider.getChunkGenerator().getWorldChunkManager();
-            List<BiomeBase> list = worldchunkmanager.a();
-            Random random = new Random(this.getSeed());
-            BlockPosition blockposition = worldchunkmanager.a(0, 0, 256, list, random);
-            ChunkCoordIntPair chunkcoordintpair = blockposition == null ? new ChunkCoordIntPair(0, 0) : new ChunkCoordIntPair(blockposition);
-
             // CraftBukkit start
             if (this.generator != null) {
                 Random rand = new Random(this.getSeed());
@@ -791,6 +785,14 @@ public class WorldServer extends World implements IAsyncTaskHandler {
             }
             // CraftBukkit end
 
+            // Paper start - this is useless if craftbukkit returns early
+            WorldChunkManager worldchunkmanager = this.chunkProvider.getChunkGenerator().getWorldChunkManager();
+            List<BiomeBase> list = worldchunkmanager.a();
+            Random random = new Random(this.getSeed());
+            BlockPosition blockposition = worldchunkmanager.a(0, 0, 256, list, random);
+            ChunkCoordIntPair chunkcoordintpair = blockposition == null ? new ChunkCoordIntPair(0, 0) : new ChunkCoordIntPair(blockposition);
+            // Paper end
+
             if (blockposition == null) {
                 WorldServer.a.warn("Unable to find spawn biome");
             }
-- 
2.22.0

